app/javascript/article-form/components/__tests__/ImageUploader.test.jsx

```diff
@@ -25,6 +33,51 @@ describe('<ImageUploader />', () => {
     expect(uploadInput.getAttribute('type')).toEqual('file');
   });
 
+  describe('when rendered in native iOS with imageUpload support', () => {
+    beforeEach(() => {
+      global.Runtime = {
+        isNativeIOS: jest.fn((namespace) => {
+          return namespace === 'imageUpload';
+        }),
+      };
+    });
+
+    it('does not display the file input', async () => {
+      const { queryByText } = render(<ImageUploader />);
+      expect(queryByText(/Upload an image/i)).not.toBeInTheDocument();
+    });
+
+    it('triggers a webkit messageHandler call when isNativeIOS', async () => {
+      global.window.webkit = {
+        messageHandlers: {
+          imageUpload: {
+            postMessage: jest.fn(),
+          },
+        },
+      };
+
+      const { queryByLabelText } = render(<ImageUploader />);
+      const uploadButton = queryByLabelText(/Upload an image/i);
+      uploadButton.click();
+      expect(
+        window.webkit.messageHandlers.imageUpload.postMessage,
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('handles a native bridge message correctly', async () => {
+      const { container, findByTitle } = render(<ImageUploader />);
+      const nativeInput = container.querySelector(
+        '#native-image-upload-message',
+      );
+
+      // Fire a change event in the hidden input with JSON payload for success
+      const fakeSuccessMessage = `{ "action": "success", "link": "/some-fake-image.jpg" }`;
+      fireEvent.change(nativeInput, { target: { value: fakeSuccessMessage } });
+
+      expect(await findByTitle(/copy markdown for image/i)).toBeDefined();
```

Comment:
Same as the other test case, this newly introduced test simulates what the native implementation would do and then checks to make sure the functions triggered correctly so that the URL is available to "copy" to clipboard.

Again, this is what I managed to get working for tests, but if you rather see this test implemented differently let me know :)